cmd/kube-dns/app/server.go

```diff
@@ -69,41 +69,28 @@ func NewKubeDNSServerDefault(config *options.KubeDNSConfig) *KubeDNSServer {
 		healthzPort:    config.HealthzPort,
 		dnsBindAddress: config.DNSBindAddress,
 		dnsPort:        config.DNSPort,
-		kd:             kdns.NewKubeDNS(kubeClient, config.ClusterDomain, config.InitialSyncTimeout, configSync),
+		kd:             dns.NewKubeDNS(kubeClient, config.ClusterDomain, config.InitialSyncTimeout, configSync),
 	}
 }
 
-// TODO: evaluate using pkg/client/clientcmd
-func newKubeClient(dnsConfig *options.KubeDNSConfig) (clientset.Interface, error) {
-	var (
-		config *restclient.Config
-		err    error
-	)
-
-	if dnsConfig.KubeMasterURL != "" && dnsConfig.KubeConfigFile == "" {
-		// Only --kube-master-url was provided.
-		config = &restclient.Config{
-			Host:          dnsConfig.KubeMasterURL,
-			ContentConfig: restclient.ContentConfig{GroupVersion: &schema.GroupVersion{Version: "v1"}},
+func newKubeClient(dnsConfig *options.KubeDNSConfig) (kubernetes.Interface, error) {
+	var config *rest.Config
+	var err error
+
+	if dnsConfig.KubeConfigFile != "" {
```

Comment:
Can you flip these so InClusterConfig is the first case - that's the normal case.